### PR TITLE
Enable continuous GeM computation.

### PR DESCRIPTION
**Pull request by maggiezimon**
The continuous calculation of GeM is enabled by adding a new score as defined in Eq. A1-A2 in ["Local-order metric for condensed-phase environments"](https://journals.aps.org/prb/abstract/10.1103/PhysRevB.97.064105).

### CODE REVIEW

File: pysages/colvars/patterns.py [hunk 1]
```diff
@@ -42,6 +44,9 @@ def __init__(
         centre_j_id,
         standard_deviation,
         mesh_size,
+        number_of_added_sites=0,
```
Comment (maggiezimon) [inline pysages/colvars/patterns.py:47]: If the continuous version of the LoM is to be used, the additional atoms should already be added at the end of the `reference` (list of coordinates of the reference structure). With `number_of_added_atoms` we specify how many were included.

**Comment by pabloferz** [comment]
@maggiezimon have you tried to take the gradient of GeM? E.g.
```python
cv = GeM(...)
jx.grad(cv.function)(positions)
```

**Comment by maggiezimon** [comment]
I didn’t run the jax.grad. Sorry!
 
I just assumed it should work as the new score function, if number_of_added_sites > 0, returns jax.lax.erfc. Maybe I need to change how we output with the calculate_lom function.

**Comment by maggiezimon** [comment]
The grad works if the positions are passed directly, instead of being read from the neighbor list.
